### ISSUE TITLE
Use of undeclared identifier 'SQLITE_OPEN_FILEPROTECTION_NONE'

### ISSUE THREAD

**Issue by monica18p** [firebase/firebase-ios-sdk#9844]
### Describe your environment

  * Xcode version: 13.4 & 13.3
  * Firebase SDK version: 9.1.0
  * Installation method: `Swift Package Manager`
  * Firebase Component: Core, Messaging (I have added FirebaseAnalytics, as there was no FirebaseCore available)
  * Target platform(s): `iOS`
  * sqlite3 --version: `3.37.0` `2021-12-09` `01:34:53` `9ff244ce0739f8ee52a3e9671adb4ee54c83c640b02e3f9d185fd2f9a179aapl`

### Describe the problem

I am getting following error on building/archiving my project
<img width="1430" alt="Screenshot 2022-05-26 at 5 38 02 PM" src="https://user-images.githubusercontent.com/6171473/170463358-abe4cb5c-54ac-40dc-b35e-e886f3dbd10f.png">

Following is the error log for your reference:
```
CompileC /Users/monica/Library/Developer/Xcode/DerivedData/eDoc-elhzszvppluiokdrpgydcxuyeqnl/Build/Intermediates.noindex/Firebase.build/UAT-iphonesimulator/FirebaseMessaging.build/Objects-normal/x86_64/FIRMessagingRmqManager.o /Users/monica/Library/Developer/Xcode/DerivedData/eDoc-elhzszvppluiokdrpgydcxuyeqnl/SourcePackages/checkouts/firebase-ios-sdk/FirebaseMessaging/Sources/FIRMessagingRmqManager.m normal x86_64 objective-c com.apple.compilers.llvm.clang.1_0.compiler (in target 'FirebaseMessaging' from project 'Firebase')
    cd /Users/monica/Library/Developer/Xcode/DerivedData/eDoc-elhzszvppluiokdrpgydcxuyeqnl/SourcePackages/checkouts/firebase-ios-sdk
    export LANG\=en_US.US-ASCII
    /Applications/Xcode13.4.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang -x objective-c -target x86_64-apple-ios11.0-simulator -fmessage-length\=0 -fdiagnostics-show-note-include-stack -fmacro-backtrace-limit\=0 -std\=c99 -fobjc-arc -fmodules -gmodules -fmodules-cache-path\=/Users/monica/Library/Developer/Xcode/DerivedData/ModuleCache.noindex -fmodules-prune-interval\=86400 -fmodules-prune-after\=345600 -fbuild-session-file\=/Users/monica/Library/Developer/Xcode/DerivedData/ModuleCache.noindex/Session.modulevalidation -fmodules-validate-once-per-build-session -Wnon-modular-include-in-framework-module -Werror\=non-modular-include-in-framework-module -fmodule-name\=FirebaseMessaging -Wno-trigraphs -fpascal-strings -Os -Wno-missing-field-initializers -Wno-missing-prototypes -Wno-return-type -Wno-implicit-atomic-properties -Wno-objc-interface-ivars -Wno-arc-repeated-use-of-weak -Wno-missing-braces -Wparentheses -Wswitch -Wno-unused-function -Wno-unused-label -Wno-unused-parameter -Wno-unused-variable -Wunused-value -Wno-empty-body -Wno-uninitialized -Wno-unknown-pragmas -Wno-shadow -Wno-four-char-constants -Wno-conversion -Wno-constant-conversion -Wno-int-conversion -Wno-bool-conversion -Wno-enum-conversion -Wno-float-conversion -Wno-non-literal-null-conversion -Wno-objc-literal-conversion -Wshorten-64-to-32 -Wpointer-sign -Wno-newline-eof -Wno-selector -Wno-strict-selector-match -Wno-undeclared-selector -Wno-deprecated-implementations -DSWIFT_PACKAGE -DPB_FIELD_32BIT\=1 -DPB_NO_PACKED_STRUCTS\=1 -DPB_ENABLE_MALLOC\=1 -DOBJC_OLD_DISPATCH_PROTOTYPES\=1 -isysroot /Applications/Xcode13.4.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator15.5.sdk -fasm-blocks -fstrict-aliasing -Wprotocol -Wdeprecated-declarations -g -Wno-sign-conversion -Wno-infinite-recursion -Wno-comma -Wno-block-capture-autoreleasing -Wno-strict-prototypes -Wno-semicolon-before-method-body -fobjc-abi-version\=2 -fobjc-legacy-dispatch -I/Users/monica/Library/Developer/Xcode/DerivedData/eDoc-elhzszvppluiokdrpgydcxuyeqnl/SourcePackages/checkouts/GoogleDataTransport/GoogleDataTransport/GDTCORLibrary/Public -I/Users/monica/Library/Developer/Xcode/DerivedData/eDoc-elhzszvppluiokdrpgydcxuyeqnl/SourcePackages/checkouts/nanopb/spm_headers -I/Users/monica/Library/Developer/Xcode/DerivedData/eDoc-elhzszvppluiokdrpgydcxuyeqnl/SourcePackages/checkouts/promises/Sources/FBLPromises/include -I/Users/monica/Library/Developer/Xcode/DerivedData/eDoc-elhzszvppluiokdrpgydcxuyeqnl/SourcePackages/checkouts/GoogleUtilities/GoogleUtilities/UserDefaults/Public -I/Users/monica/Library/Developer/Xcode/DerivedData/eDoc-elhzszvppluiokdrpgydcxuyeqnl/SourcePackages/checkouts/GoogleUtilities/GoogleUtilities/Network/Public -I/Users/monica/Library/Developer/Xcode/DerivedData/eDoc-elhzszvppluiokdrpgydcxuyeqnl/SourcePackages/checkouts/GoogleUtilities/GoogleUtilities/NSData+zlib/Public -I/Users/monica/Library/Developer/Xcode/DerivedData/eDoc-elhzszvppluiokdrpgydcxuyeqnl/SourcePackages/checkouts/GoogleUtilities/GoogleUtilities/AppDelegateSwizzler/Public -I/Users/monica/Library/Developer/Xcode/DerivedData/eDoc-elhzszvppluiokdrpgydcxuyeqnl/SourcePackages/checkouts/GoogleUtilities/GoogleUtilities/Logger/Public -I/Users/monica/Library/Developer/Xcode/DerivedData/eDoc-elhzszvppluiokdrpgydcxuyeqnl/SourcePackages/checkouts/GoogleUtilities/GoogleUtilities/Environment/Public -I/Users/monica/Library/Developer/Xcode/DerivedData/eDoc-elhzszvppluiokdrpgydcxuyeqnl/SourcePackages/checkouts/GoogleUtilities/GoogleUtilities/Reachability/Public -I/Users/monica/Library/Developer/Xcode/DerivedData/eDoc-elhzszvppluiokdrpgydcxuyeqnl/SourcePackages/checkouts/firebase-ios-sdk/FirebaseInstallations/Source/Library/Public -I/Users/monica/Library/Developer/Xcode/DerivedData/eDoc-elhzszvppluiokdrpgydcxuyeqnl/SourcePackages/checkouts/firebase-ios-sdk/FirebaseCore/Sources/Public -I/Users/monica/Library/Developer/Xcode/DerivedData/eDoc-elhzszvppluiokdrpgydcxuyeqnl/SourcePackages/checkouts/firebase-ios-sdk/Firebase/CoreDiagnostics/FIRCDLibrary -I/Users/monica/Library/Developer/Xcode/DerivedData/eDoc-elhzszvppluiokdrpgydcxuyeqnl/SourcePackages/checkouts/firebase-ios-sdk/CoreOnly/Sources -I/Users/monica/Library/Developer/Xcode/DerivedData/eDoc-elhzszvppluiokdrpgydcxuyeqnl/Build/Products/UAT-iphonesimulator/include -I/Users/monica/Library/Developer/Xcode/DerivedData/eDoc-elhzszvppluiokdrpgydcxuyeqnl/SourcePackages/checkouts/firebase-ios-sdk/FirebaseMessaging/Sources/Public -I/Users/monica/Library/Developer/Xcode/DerivedData/eDoc-elhzszvppluiokdrpgydcxuyeqnl/SourcePackages/checkouts/firebase-ios-sdk -I/Users/monica/Library/Developer/Xcode/DerivedData/eDoc-elhzszvppluiokdrpgydcxuyeqnl/Build/Intermediates.noindex/Firebase.build/UAT-iphonesimulator/FirebaseMessaging.build/DerivedSources-normal/x86_64 -I/Users/monica/Library/Developer/Xcode/DerivedData/eDoc-elhzszvppluiokdrpgydcxuyeqnl/Build/Intermediates.noindex/Firebase.build/UAT-iphonesimulator/FirebaseMessaging.build/DerivedSources/x86_64 -I/Users/monica/Library/Developer/Xcode/DerivedData/eDoc-elhzszvppluiokdrpgydcxuyeqnl/Build/Intermediates.noindex/Firebase.build/UAT-iphonesimulator/FirebaseMessaging.build/DerivedSources -F/Users/monica/Library/Developer/Xcode/DerivedData/eDoc-elhzszvppluiokdrpgydcxuyeqnl/Build/Products/UAT-iphonesimulator/PackageFrameworks -F/Users/monica/Library/Developer/Xcode/DerivedData/eDoc-elhzszvppluiokdrpgydcxuyeqnl/Build/Products/UAT-iphonesimulator/PackageFrameworks -F/Users/monica/Library/Developer/Xcode/DerivedData/eDoc-elhzszvppluiokdrpgydcxuyeqnl/Build/Products/UAT-iphonesimulator/PackageFrameworks -F/Users/monica/Library/Developer/Xcode/DerivedData/eDoc-elhzszvppluiokdrpgydcxuyeqnl/Build/Products/UAT-iphonesimulator/PackageFrameworks -F/Users/monica/Library/Developer/Xcode/DerivedData/eDoc-elhzszvppluiokdrpgydcxuyeqnl/Build/Products/UAT-iphonesimulator/PackageFrameworks -F/Users/monica/Library/Developer/Xcode/DerivedData/eDoc-elhzszvppluiokdrpgydcxuyeqnl/Build/Products/UAT-iphonesimulator/PackageFrameworks -F/Users/monica/Library/Developer/Xcode/DerivedData/eDoc-elhzszvppluiokdrpgydcxuyeqnl/Build/Products/UAT-iphonesimulator/PackageFrameworks -F/Users/monica/Library/Developer/Xcode/DerivedData/eDoc-elhzszvppluiokdrpgydcxuyeqnl/Build/Products/UAT-iphonesimulator/PackageFrameworks -F/Users/monica/Library/Developer/Xcode/DerivedData/eDoc-elhzszvppluiokdrpgydcxuyeqnl/Build/Products/UAT-iphonesimulator/PackageFrameworks -F/Users/monica/Library/Developer/Xcode/DerivedData/eDoc-elhzszvppluiokdrpgydcxuyeqnl/Build/Products/UAT-iphonesimulator/PackageFrameworks -F/Users/monica/Library/Developer/Xcode/DerivedData/eDoc-elhzszvppluiokdrpgydcxuyeqnl/Build/Products/UAT-iphonesimulator/PackageFrameworks -F/Users/monica/Library/Developer/Xcode/DerivedData/eDoc-elhzszvppluiokdrpgydcxuyeqnl/Build/Products/UAT-iphonesimulator/PackageFrameworks -F/Users/monica/Library/Developer/Xcode/DerivedData/eDoc-elhzszvppluiokdrpgydcxuyeqnl/Build/Products/UAT-iphonesimulator/PackageFrameworks -F/Users/monica/Library/Developer/Xcode/DerivedData/eDoc-elhzszvppluiokdrpgydcxuyeqnl/Build/Products/UAT-iphonesimulator/PackageFrameworks -F/Users/monica/Library/Developer/Xcode/DerivedData/eDoc-elhzszvppluiokdrpgydcxuyeqnl/Build/Products/UAT-iphonesimulator/PackageFrameworks -F/Users/monica/Library/Developer/Xcode/DerivedData/eDoc-elhzszvppluiokdrpgydcxuyeqnl/Build/Products/UAT-iphonesimulator -F/Applications/Xcode13.4.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/Library/Frameworks -iframework /Applications/Xcode13.4.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator15.5.sdk/Developer/Library/Frameworks -fmodule-map-file\=/Users/monica/Library/Developer/Xcode/DerivedData/eDoc-elhzszvppluiokdrpgydcxuyeqnl/Build/Intermediates.noindex/GeneratedModuleMaps-iphonesimulator/GoogleDataTransport.modulemap -fmodule-map-file\=/Users/monica/Library/Developer/Xcode/DerivedData/eDoc-elhzszvppluiokdrpgydcxuyeqnl/Build/Intermediates.noindex/GeneratedModuleMaps-iphonesimulator/nanopb.modulemap -fmodule-map-file\=/Users/monica/Library/Developer/Xcode/DerivedData/eDoc-elhzszvppluiokdrpgydcxuyeqnl/Build/Intermediates.noindex/GeneratedModuleMaps-iphonesimulator/GoogleUtilities_UserDefaults.modulemap -fmodule-map-file\=/Users/monica/Library/Developer/Xcode/DerivedData/eDoc-elhzszvppluiokdrpgydcxuyeqnl/Build/Intermediates.noindex/GeneratedModuleMaps-iphonesimulator/GoogleUtilities_Network.modulemap -fmodule-map-file\=/Users/monica/Library/Developer/Xcode/DerivedData/eDoc-elhzszvppluiokdrpgydcxuyeqnl/Build/Intermediates.noindex/GeneratedModuleMaps-iphonesimulator/GoogleUtilities_NSData.modulemap -fmodule-map-file\=/Users/monica/Library/Developer/Xcode/DerivedData/eDoc-elhzszvppluiokdrpgydcxuyeqnl/Build/Intermediates.noindex/GeneratedModuleMaps-iphonesimulator/GoogleUtilities_AppDelegateSwizzler.modulemap -fmodule-map-file\=/Users/monica/Library/Developer/Xcode/DerivedData/eDoc-elhzszvppluiokdrpgydcxuyeqnl/Build/Intermediates.noindex/GeneratedModuleMaps-iphonesimulator/GoogleUtilities_Logger.modulemap -fmodule-map-file\=/Users/monica/Library/Developer/Xcode/DerivedData/eDoc-elhzszvppluiokdrpgydcxuyeqnl/Build/Intermediates.noindex/GeneratedModuleMaps-iphonesimulator/GoogleUtilities_Environment.modulemap -fmodule-map-file\=/Users/monica/Library/Developer/Xcode/DerivedData/eDoc-elhzszvppluiokdrpgydcxuyeqnl/Build/Intermediates.noindex/GeneratedModuleMaps-iphonesimulator/GoogleUtilities_Reachability.modulemap -fmodule-map-file\=/Users/monica/Library/Developer/Xcode/DerivedData/eDoc-elhzszvppluiokdrpgydcxuyeqnl/Build/Intermediates.noindex/GeneratedModuleMaps-iphonesimulator/FirebaseInstallations.modulemap -fmodule-map-file\=/Users/monica/Library/Developer/Xcode/DerivedData/eDoc-elhzszvppluiokdrpgydcxuyeqnl/Build/Intermediates.noindex/GeneratedModuleMaps-iphonesimulator/FirebaseCore.modulemap -fmodule-map-file\=/Users/monica/Library/Developer/Xcode/DerivedData/eDoc-elhzszvppluiokdrpgydcxuyeqnl/Build/Intermediates.noindex/GeneratedModuleMaps-iphonesimulator/FirebaseCoreInternal.modulemap -fmodule-map-file\=/Users/monica/Library/Developer/Xcode/DerivedData/eDoc-elhzszvppluiokdrpgydcxuyeqnl/Build/Intermediates.noindex/GeneratedModuleMaps-iphonesimulator/FirebaseCoreDiagnostics.modulemap -DXcode -MMD -MT dependencies -MF /Users/monica/Library/Developer/Xcode/DerivedData/eDoc-elhzszvppluiokdrpgydcxuyeqnl/Build/Intermediates.noindex/Firebase.build/UAT-iphonesimulator/FirebaseMessaging.build/Objects-normal/x86_64/FIRMessagingRmqManager.d --serialize-diagnostics /Users/monica/Library/Developer/Xcode/DerivedData/eDoc-elhzszvppluiokdrpgydcxuyeqnl/Build/Intermediates.noindex/Firebase.build/UAT-iphonesimulator/FirebaseMessaging.build/Objects-normal/x86_64/FIRMessagingRmqManager.dia -c /Users/monica/Library/Developer/Xcode/DerivedData/eDoc-elhzszvppluiokdrpgydcxuyeqnl/SourcePackages/checkouts/firebase-ios-sdk/FirebaseMessaging/Sources/FIRMessagingRmqManager.m -o /Users/monica/Library/Developer/Xcode/DerivedData/eDoc-elhzszvppluiokdrpgydcxuyeqnl/Build/Intermediates.noindex/Firebase.build/UAT-iphonesimulator/FirebaseMessaging.build/Objects-normal/x86_64/FIRMessagingRmqManager.o

/Users/monica/Library/Developer/Xcode/DerivedData/eDoc-elhzszvppluiokdrpgydcxuyeqnl/SourcePackages/checkouts/firebase-ios-sdk/FirebaseMessaging/Sources/FIRMessagingRmqManager.m:495:56: error: use of undeclared identifier 'SQLITE_OPEN_FILEPROTECTION_NONE'
          SQLITE_OPEN_READWRITE | SQLITE_OPEN_CREATE | SQLITE_OPEN_FILEPROTECTION_NONE, NULL);
                                                       ^
/Users/monica/Library/Developer/Xcode/DerivedData/eDoc-elhzszvppluiokdrpgydcxuyeqnl/SourcePackages/checkouts/firebase-ios-sdk/FirebaseMessaging/Sources/FIRMessagingRmqManager.m:514:56: error: use of undeclared identifier 'SQLITE_OPEN_FILEPROTECTION_NONE'
          SQLITE_OPEN_READWRITE | SQLITE_OPEN_CREATE | SQLITE_OPEN_FILEPROTECTION_NONE, NULL);
                                                       ^
2 errors generated.
```

I have already tried clearing the derivedData, reset package cache, reinstall firebase package.
Kindly assist to fix the issue.

**Comment by rizafran** [comment]
Thanks for reporting, @monica18p. I'm unable to reproduce the issue. Just want to confirm if you have selected the version instead of Branch in the `Dependency Rule` (see screenshot below)?

<img width="844" alt="Screen Shot 2022-05-26 at 8 22 08 PM" src="https://user-images.githubusercontent.com/56452638/170488416-11929e55-fd38-4a15-884e-34dc1c36e419.png">

**Comment by paulb777** [comment]
Also, please make sure that the app is using the system version of sqlite.  Right clicking on SQLITE_OPEN_FILEPROTECTION_NONE  to see the definition shows it here:

![Screen Shot 2022-05-26 at 6 21 45 AM](https://user-images.githubusercontent.com/73870/170496266-5dad43cc-4d41-411f-aaa1-081fadec85e6.png)

**Comment by monica18p** [comment]
Hi ,
Thanks @rizafran & @paulb777  for prompt responses.
I got this issue when I moved the header files before compile sources in my project's submodules. 
I am not sure why this change triggered this issue. For now I have revert my header file change and it has fixed the firebase issue. Therefore, I will close this issue now. In case I come across some other findings i will comment down in this thread.

Thanks for the support 🙏
